pkg/controllers/provisioning/scheduling/topologygroup.go

```diff
@@ -172,25 +171,45 @@ func (t *TopologyGroup) Hash() uint64 {
 // nextDomainTopologySpread returns a scheduling.Requirement that includes a node domain that a pod should be scheduled to.
 // If there are multiple eligible domains, we return any random domain that satisfies the `maxSkew` configuration.
 // If there are no eligible domains, we return a `DoesNotExist` requirement, implying that we could not satisfy the topologySpread requirement.
+// nolint:gocyclo
 func (t *TopologyGroup) nextDomainTopologySpread(pod *v1.Pod, podDomains, nodeDomains *scheduling.Requirement) *scheduling.Requirement {
 	// min count is calculated across all domains
 	min := t.domainMinCount(podDomains)
 	selfSelecting := t.selects(pod)
 
 	minDomain := ""
 	minCount := int32(math.MaxInt32)
-	for domain := range t.domains {
-		// but we can only choose from the node domains
-		if nodeDomains.Has(domain) {
-			// comment from kube-scheduler regarding the viable choices to schedule to based on skew is:
-			// 'existing matching num' + 'if self-match (1 or 0)' - 'global min matching num' <= 'maxSkew'
-			count := t.domains[domain]
-			if selfSelecting {
-				count++
+
+	// If we are explicitly selecting on specific node domains ("In" requirement) and the number of node domains
+	// is less than our total domains, this is going to be more efficient to iterate through
+	// This is particularly useful when considering the hostname topology key that can have a
+	// lot of t.domains but only a single nodeDomain
+	if nodeDomains.Operator() == v1.NodeSelectorOpIn && nodeDomains.Len() < len(t.domains) {
```

Comment:
Ah, you're saying because we calculate available domains at the beginning based on the possibilities on the nodepool that this is moot -- think I agree